### ISSUE TITLE
Colour vision deficiency issues

### ISSUE THREAD

**Issue by 06nenglish** [vlsi/ksar#107]
Hi,

Really love the tool, going to help a great deal when performing tests. However I have one slight issue, maybe a quick way round maybe not. 

I'm colour blind with Red, Green and Blue and struggle with similar colours quite badly and as you can imagine the graphs are very hard to decipher which colour matches the line on the graph.

If there is no option (I can't see anything set anywhere), it's something I will look into contributing as I know it'll help out many users.

Thanks!

**Comment by vlsi** [comment]
I know some colors (all of them?) are hard-coded in https://github.com/vlsi/ksar/blob/master/src/main/resources/Config.xml

I know there's http://colorbrewer2.org/  @06nenglish , do you think it is relevant?

**Comment by Pitterling** [comment]
Yes, its hard-coded, but only few graphs are covered. 

The ksar_all.log file lists all missing colour code (you need to open each charts)

> 2017-08-19 13:45:48.970 [AWT-EventQueue-0] WARN  net.atomique.ksar.GlobalOptions - color not found for tag %irq
> 2017-08-19 13:45:48.970 [AWT-EventQueue-0] WARN  net.atomique.ksar.GlobalOptions - color not found for tag %soft
> 2017-08-19 13:45:48.970 [AWT-EventQueue-0] WARN  net.atomique.ksar.GlobalOptions - color not found for tag %guest
> 2017-08-19 13:45:55.032 [AWT-EventQueue-0] WARN  net.atomique.ksar.GlobalOptions - color not found for tag %irq
> 2017-08-19 13:45:55.032 [AWT-EventQueue-0] WARN  net.atomique.ksar.GlobalOptions - color not found for tag %soft
> 2017-08-19 13:45:55.032 [AWT-EventQueue-0] WARN  net.atomique.ksar.GlobalOptions - color not found for tag %guest
> 2017-08-19 13:46:05.984 [AWT-EventQueue-0] WARN  net.atomique.ksar.GlobalOptions - color not found for tag kbmemfree

Feel free to create a "colour-blind" Config.xml and we make this available in the options dialog

**Comment by Pitterling** [comment]
I was playing with this colours

![image](https://user-images.githubusercontent.com/4181164/29499203-2ffbc2a0-860c-11e7-80a0-a5dde573e5bb.png)

and adjusted the CPU chart, looks like this now
![image](https://user-images.githubusercontent.com/4181164/29499208-57acda64-860c-11e7-9fc7-52142ffe10e7.png)

 @06nenglish what do you think?

**Comment by 06nenglish** [comment]
Apologies for the slow responses both!

@Pitterling that is a lot easier for me to understand. I really appreciate you taking the time to look into it for me. Was this done via the config.xml or was it actually code that required changing?

**Comment by vlsi** [comment]
@Pitterling , `%guest` is likely too light (it is hard to distinguish from white).
Is there a logic for `sys`, `soft`, `steal` being "red" and `usr, nice, iowait` being "blue"?

Diverging scale might suit better here.

**Comment by Pitterling** [comment]
@06nenglish yes, i only changed config.xml, i just picked the 8 proposed colours and distributed them across the stats ...

```
       <itemcolor name="%usr">
            <color>116,173,209</color>
        </itemcolor>
        <itemcolor name="%sys">
            <color>215,48,39</color>
        </itemcolor>
...
        <itemcolor name="%nice">
            <color>69,117,180</color>
        </itemcolor>
        <itemcolor name="%steal">
            <color>253,174,97</color>
        </itemcolor>
        <itemcolor name="%irq">
            <color>254,224,144</color>
        </itemcolor>
        <itemcolor name="%soft">
            <color>244,109,67</color>
        </itemcolor>
        <itemcolor name="%guest">
            <color>224,243,248</color>
        </itemcolor>
        <itemcolor name="%iowait">
            <color>171,217,233</color>
        </itemcolor>
```

@ vlsi I'm open to any change .. finally the colour code scheme should come from colour blind people ...
I consider sys, soft, steal as more important .. therefore i wanted to highlight them more .. the light colours are proposed by the tool ..

In general we could offer, several sets .. default, colour blind ... etc

**Comment by Pitterling** [comment]
@06nenglish are you still looking into this? I expected an adjusted config.xml file, coming from color-blind affected people ...
